tests/test_trace_exporter.py

```diff
@@ -0,0 +1,123 @@
+import unittest
+
+from unittest.mock import patch
+
+
+from opentelemetry import trace as trace_api
+from opentelemetry.sdk import trace
+from opentelemetry.ext.lightstep import LightStepSpanExporter
+from opentelemetry.sdk.trace.export import SpanExportResult
+
+
+class MockTracer:
+    """ Used to mock lightstep.Tracer """
+
+    def __init__(
+        self,
+        component_name,
+        access_token,
+        collector_host,
+        collector_port,
+        collector_encryption,
+        verbosity,
+        use_http=False,
+        use_thrift=False,
+    ):
+        self.name = component_name
+        self.token = access_token
+        self.host = collector_host
+        self.port = collector_port
+        self.encryption = collector_encryption
+        self.verbosity = verbosity
+        self.use_http = use_http
+        self.use_thrift = use_thrift
+        self.spans = []
+
+    def record(self, span):
+        self.spans.append(span)
+
+    def flush(self):
+        pass
+
+
+class TestLightStepSpanExporter(unittest.TestCase):
+    @patch("lightstep.Tracer", MockTracer)
```

Comment:
It is a common code smell to end up creating a "mock" something (for example a class or a function) from scratch. That should be the job of the standard mock library. For example here, instead of declaring a `MockTracer` class, just use `patch` and pass the mock to the test case:

```python
@patch("lightstep.Tracer")
def test_constructor_default(self, mock_tracer):
    # Configure mock_tracer here, for example, define the behavior for record and flush
```
If you want to use the same mock object in several test cases, then you can use `setUp` and `tearDown` to add the mock as a `self` attribute (making it available for all test cases) and to control the patching and unpatching of the `lightstep.Tracer` class:

```python
def setUp(self):
    self.mock_tracer = patch("lightstep.Tracer")
    self.mock_tracer.start()

def tearDown(self):
    self.mock_tracer.stop()
```

Reply:
Cleaned up. Thanks for the help!